sdk/java/pulumi/src/main/java/com/pulumi/test/internal/PulumiTestInternal.java

```diff
@@ -42,4 +147,274 @@ public CompletableFuture<TestResult> runTestAsync(
                 this.stackContext.exports()
         ));
     }
+
+    /**
+     * @see PulumiTest#cleanup()
+     */
+    public static void cleanup() {
+        // ensure we don't get the error:
+        //   java.lang.IllegalStateException: Deployment.getInstance should only be set once at the beginning of a 'run' call.
+        DeploymentImpl.internalUnsafeDestroyInstance();
+    }
+
+    @InternalUse
+    @CanIgnoreReturnValue
+    public static <T> OutputData<T> extractOutputData(Output<T> output) {
+        return Internal.of(output).getDataAsync().join();
+    }
+
+    @InternalUse
+    public static final class APIInternal implements PulumiTest.API {
```

Comment:
this is where we combine the builder with the operations for a DSL used as part of the top level API